sql/analyzer/stored_procedures.go

```diff
@@ -224,6 +224,22 @@ func resolveProcedureParams(paramNames map[string]struct{}, proc sql.Node) (sql.
 	if err != nil {
 		return nil, err
 	}
+	newProcNode, err = plan.TransformUp(newProcNode, func(n sql.Node) (sql.Node, error) {
```

Comment:
You should add a note on why this is necessary: because the Source of inserts is analyzed separately from the rest of the query